census/estimator/trainer/task.py

```diff
@@ -0,0 +1,154 @@
+import argparse
+
+import model
+
+import tensorflow as tf
+from tensorflow.contrib.learn.python.learn import learn_runner
+from tensorflow.contrib.learn.python.learn.utils import (
+    saved_model_export_utils)
+
+
+def generate_experiment_fn(train_file,
+                           eval_file,
+                           num_epochs=None,
+                           train_batch_size=40,
+                           eval_batch_size=40,
+                           embedding_size=8,
+                           first_layer_size=100,
+                           num_layers=4,
+                           scale_factor=0.7,
+                           **experiment_args):
+  def _experiment_fn(output_dir):
+    train_input = model.generate_input_fn(
+        train_file,
+        num_epochs=num_epochs,
+        batch_size=train_batch_size,
+        skip_header_lines=model.TRAIN_HEADER_LINES
+    )
+    eval_input = model.generate_input_fn(
+        eval_file,
+        batch_size=eval_batch_size,
+        skip_header_lines=model.EVAL_HEADER_LINES,
+        shuffle=False
+    )
+    return tf.contrib.learn.Experiment(
+        model.build_estimator(
+            job_dir,
+            embedding_size=embedding_size,
+            hidden_units=[
+                int(first_layer_size * scale_factor**i)
+                for i in range(num_layers)
+            ]
+        ),
+        train_input_fn=train_input,
+        eval_input_fn=eval_input,
```

Comment:
We originally removed the eval metrics section here. However I'm not sure this is the right move, if you take a look at the eval metrics returned in the default dict, they say things like auc = 1.0 even when accuracy is 82%. Worth a double check here